### ISSUE TITLE
🐛 Skills picker

### ISSUE THREAD

**Issue by BrookeCameron** [GCTC-NTGC/gc-digital-talent#13253]
## 🐛 Bug

Tested with adding a Government work experience 

Issue with typing the exact name and hitting enter the skill is not added - even if it is an exact match. 

Particular issue with skill "Verbal Communication" 

If you hit enter with the skill typed in you get sent back to experiences and receive a success message but the skill has not been added to the experience. 

### 🛡️ Fable test results 

Fable tester comments: 
After entering communication and filtering it to verbal communication, I was unable to find one of the required skills [Verbal Communication] in the instructions. 

## 🦋 Expected Behaviour

- Users know when to add a skill
- User knows when results are being shown
- skills can be added by hitting enter if they are an exact match

## 🕵️ Details

- **Operating System**: Mac OS
- **Browser**: Safari
- **Assistive tech**: Voice over 

Was able to reproduce the issue with 

- **Operating System**: Mac OS Sequoia Version 15.3.1 
- **Browser**: Chrome Version 135.0.7049.85
- **Assistive tech**: Keyboard nav (tabs/arrows/enter)

## 🎥 Recording

https://github.com/user-attachments/assets/8f6606dc-6369-4ab9-9c31-a39b80638d2f

**Comment by petertgiles** [comment]
This looks like two separate things to me:
1. The user typed in an exact match and expected it to be selected when pressing enter.
2. Submitting the "link skill to experience" form without a skill closed the dialog and popped a success toast.

(2) seems like a clear bug that we can fix right away.  (1) I'm not so sure about.  This may be intended behaviour - an item is only selected when cursoring down through the list of matches.  That's more of an Eric+Josh question, I think.

**Comment by petertgiles** [comment]
I've scoped out the "pressing enter submited the form outside the dialog" into its own issue since that seemed pretty straightforward.
https://github.com/GCTC-NTGC/gc-digital-talent/issues/13284

**Comment by BrookeCameron** [comment]
@esizer @JoshBeveridge thoughts on this?

**Comment by esizer** [comment]
> (1) I'm not so sure about. This may be intended behaviour - an item is only selected when cursoring down through the list of matches.

@BrookeCameron This is expected behaviour. Though, I agree it is a little odd, if you aren't aware how this works and type in an exact match, I can see how people would expect submitting it would work.

I wonder if there is a way to select it if there is an exact match 🤔

**Comment by petertgiles** [comment]
The form handling bug has been solved for this dialog.  If you're interested in adding new behaviour to the combobox component, please open a new issue.
